src/components/heatmap/heatmap.tsx

```diff
@@ -84,32 +91,48 @@ const Heatmap = ({
-      const reference = response.contextualize.compact;
-      verifyConfiguration(reference);
-      if (!configErrorsPresent) {
-        const sortBy = typeof Object(config).data.sortBy !== 'undefined' ? Object(config).data.sortBy : [];
-        const ref = reference.sort(sortBy);
-        setHeader((prevHeader: any) => ({ ...prevHeader, action: 'main' }));
-        const pcid = getQueryParams(location.href).pcid
-        const ppid = getQueryParams(location.href).ppid
-        if (pcid || ppid){
-          let tempHeader = {...header}
-          if(pcid)
-            tempHeader.pcid=pcid
-          if(ppid)
-            tempHeader.ppid=ppid
-          setHeader(() => (tempHeader));
+    const geneId = location.hash.split('/')[2];
+    // try to fetch the resolver link to see if the path resolves before sending the user
+    ConfigService.http.get(`/ermrest/catalog/2/entity/Common:Gene/${geneId}`,
+      { headers: header }).then((response: any) => {
+        setIsLoading((prevLoading) => !prevLoading);
+        if (response.data?.length > 0) {
+          if (response.data[0].Array_Data) {
+            ConfigService.ERMrest.resolve(ermrestURI.ermrestUri, header).then((response: any) => {
+              const reference = response.contextualize.compact;
+              verifyConfiguration(reference);
+              if (!configErrorsPresent) {
+                const sortBy = typeof Object(config).data.sortBy !== 'undefined' ? Object(config).data.sortBy : [];
+                const ref = reference.sort(sortBy);
+                setHeader((prevHeader: any) => ({ ...prevHeader, action: 'main' }));
+                const pcid = getQueryParams(location.href).pcid
+                const ppid = getQueryParams(location.href).ppid
+                if (pcid || ppid) {
+                  let tempHeader = { ...header }
+                  if (pcid)
+                    tempHeader.pcid = pcid
+                  if (ppid)
+                    tempHeader.ppid = ppid
+                  setHeader(() => (tempHeader));
+                }
+                ref.read(1000, header).then((page: any) => {
+                  readAll(page);
+                }).catch(() => {
+                  setInvalidConfigs(['Error while reading data from Ermrestjs']);
+                  setConfigErrorsPresent(true);
+                });
+              }
+            });
+          } else {
+            throw new ChaiseError('Data not available', 'No heatmap data available for the given Id<br>');
+          }
+        } else {
+          throw new ChaiseError('Record not found', 'No matching record found for the given Id<br>');
         }
-        ref.read(1000, header).then((page: any) => {
-          readAll(page);
-        }).catch(() => {
-          setInvalidConfigs(['Error while reading data from Ermrestjs']);
-          setConfigErrorsPresent(true);
-        });
-      }
-    });
+      }).catch((err: any) => {
+        setNoData(() => true);
```

Comment:
We should do `setIsLoading(false)` in this catch block before throwing an error since we know the app is done setting up.

Reply:
I still need something to know that data is not there so the page return empty fragment